src/App.js

```diff
@@ -1,9 +1,99 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import Form from './components/Form';
+import Todo from './components/Todo';
+import Done from './components/Done';
+import GlobalStyle from './styles/GloalStyle';
+
+const initialTodoData = localStorage.getItem('list')
+  ? JSON.parse(localStorage.getItem('list'))
+  : [];
+
 function App() {
+  const [list, setList] = useState(initialTodoData);
+  const [value, setValue] = useState('');
+
+  const getTodo = (todo) => {
+    const newTodo = {
+      id: Date.now(),
+      title: todo,
+      completed: false,
+    };
+    setList([...list, newTodo]);
+    localStorage.setItem('list', JSON.stringify([...list, newTodo]));
+  };
+
+  const toggleTodo = (id) => {
+    let toggledTodo = list.map((data) =>
+      data.id === id ? { ...data, completed: !data.completed } : data
+    );
+    setList(toggledTodo);
+    localStorage.setItem('list', JSON.stringify(toggledTodo));
+  };
+
+  const deleteTodo = (id) => {
+    let deletedTodo = list.filter((data) => data.id !== id);
+    setList(deletedTodo);
+    localStorage.setItem('list', JSON.stringify(deletedTodo));
+  };
+
+  let countTodo = 0;
+  list.map((data) => (data.completed ? data : countTodo++));
```

Comment:
요기는 filter사용해도 좋을 것 같아요
별 차이가 없기는 하지만, map은 약간 `각각의 원소에 작업을 해주고 싶을 때`사용하고, filter은 `특정한 data를 가진 원소를 걸러내고싶을 때` 사용하는 것 같아서요!
큰차이는 없지만, [filter와 map의 performance를 비교한 글](https://daesuni.github.io/Loop-performance/)을 놓고 갑니다..!
```suggestion
let countTodo = list.filter((data) => {data.completed}).length;
```

Reply:
저도 좋은 정보 알아갑니다!👍👍